src/gui/qgsdockablewidgethelper.h

```diff
@@ -50,17 +54,64 @@ class GUI_EXPORT QgsNonRejectableDialog : public QDialog
  */
 class GUI_EXPORT QgsDockableWidgetHelper : public QObject
 {
+    static inline QgsSettingsTreeNode *sTtreeDockConfigs = QgsGui::sTtreeWidgetGeometry->createNamedListNode( QStringLiteral( "docks" ) ) SIP_SKIP;
+
+    static const QgsSettingsEntryBool *sSettingsIsDocked SIP_SKIP;
+    static const QgsSettingsEntryVariant *sSettingsWindowGeometry SIP_SKIP;
+    static const QgsSettingsEntryEnumFlag<Qt::DockWidgetArea> *sSettingsDockArea SIP_SKIP;
+
     Q_OBJECT
   public:
+    enum class OpeningMode : int
+    {
+      RespectSetting, //! Respect the setting used
+      ForceDocked,    //! Force the widget to be docked, despite its settings
+      ForceDialog,    //! Force the widget to be shown in a dialog, despite its settings
+    };
+
+    enum class Option : int
+    {
+      RaiseTab = 1 << 1,        //!< Raise Tab
+      PermanentWidget = 1 << 2, //!< The widget (either as a dock or window) cannot be destroyed and must be hidden instead
+    };
+    Q_ENUM( Option )
+    Q_DECLARE_FLAGS( Options, Option )
+
     /**
      * Constructs an object that is responsible of making a docked widget or a window titled \a windowTitle that holds the \a widget
      * The ownership of \a widget is returned to \a ownerWindow once the object is destroyed.
      *
      * If \a usePersistentWidget is TRUE then the \a widget (either as a dock or window) cannot be destroyed and must be hidden instead.
      */
     QgsDockableWidgetHelper( bool isDocked, const QString &windowTitle, QWidget *widget, QMainWindow *ownerWindow, Qt::DockWidgetArea defaultDockArea = Qt::NoDockWidgetArea, const QStringList &tabifyWith = QStringList(), bool raiseTab = false, const QString &windowGeometrySettingsKey = QString(), bool usePersistentWidget = false );
```

Comment:
Do want to deprecate this function?

Reply:
Since it's a private API, I would then probably just remove it and simplify the code a bit more

Reply:
@3nids , you want to do it in this PR or a follow up one? I assume you'll have to migrate 2D/3D map view et al for that.

Reply:
now done in this PR